pending-documentation.md

```diff
@@ -107,6 +107,338 @@ argument and parameter are K, and it has no effect on the resulting sort of
+not be heated unless `isKResult(BE2)` is true, meaning that `BE2` must be
+evaluated first):
+
+```k
+rule <k>  BE1:BExp || HOLE:BExp => HOLE ~> BE1 ||  [] ... </k> [heat]
+rule <k> HOLE:BExp ||  BE2:BExp => HOLE ~>  [] || BE2 ... </k> requires isKResult(BE2) [heat]
+
+rule <k> HOLE:BExp ~>  [] || BE2 => HOLE ||  BE2 ... </k> [cool]
+rule <k> HOLE:BExp ~> BE1 ||  [] =>  BE1 || HOLE ... </k> [cool]
+```
+
+### Context Declaration
+
+Sometimes more advanced evaluation strategies are needed. By default, the
+`strict` and `seqstrict` attributes are limited in that they cannot describe
+the _context_ in which heating or cooling should occur. When this type of
+control over the evaluation strategy is required, `context` sentences can be
+used to simplify the process of declaring heating and cooling when it would be
+unnecessarily verbose to write heating and cooling rules directly.
+
+For example, if the user wants to heat a term if it exists under a `foo`
+constructor if the term to be heated is of sort `bar`, one might write the
+following context:
+
+```k
+context foo(HOLE:Bar)
+```
+
+Once again, note that `HOLE` is just a variable, but one that has special
+meaning to the compiler indicating the position in the context that should
+be heated or cooled.
+
+This will automatically generate the following sentences:
+
+```k
+rule <k> foo(HOLE:Bar) => HOLE ~> foo([]) ... </k> [heat]
+rule <k> HOLE:Bar ~> foo([]) => foo(HOLE) ... </k> [cool]
+```
+
+The user may also write the K cell explicitly in the context declaration
+if they want to match on another cell as well, for example:
+
+```k
+context <k> foo(HOLE:Bar) ... </kl> <state> .Map </state>
+```
+
+This context will now only heat or cool if the `state` cell is empty.
+
+### Side conditions in context declarations
+
+The user is allowed to write a side condition in a context declaration, like
+so:
+
+```k
+context foo(HOLE:Bar) requires baz(HOLE)
+```
+
+This side condition will be appended verbatim to the heating rule that is
+generated, however, it will not affect the cooling rule that is generated:
+
+```k
+rule <k> foo(HOLE:Bar) => HOLE ~> foo([]) ... </k> requirese baz(HOLE) [heat]
+rule <k> HOLE:Bar ~> foo([]) => foo(HOLE) ... </k> [cool]
+```
+
+### Rewrites in context declarations
+
+The user can also include exactly one rewrite operation in a context
+declaration if that rule rewrites the variable `HOLE` on the left hand side
+to a term containing `HOLE` on the right hand side. For exampl;e:
+
+```k
+context foo(HOLE:Bar => bar(HOLE))
+```
+
+In this case, the code generated will be as follows:
+
+```k
+rule <k> foo(HOLE:Bar) => bar(HOLE) ~> foo([]) ... </k> [heat]
+rule <k> bar(HOLE:Bar) ~> foo([]) => foo(HOLE) ... </k> [cool]
+```
+
+This can be useful if the user wishes to evaluate a term using a different
+set of rules than normal. For example, if the user wants to evaluate a term
+to an rvalue instead of an lvalue.
```

Comment:
`rvalue` and `lvalue` are completely out of nowhere here, probably better to just remove this sentence rather than send the user searching for "what is `rvalue`.

Also, if `rvalue` and `lvalue` are supposed to be actual sorts (which I assume, but I don't know) then capitalize them correctly and put them in code brackets. But if you keep them, explain them.